smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/integration/http-api-key-auth.ts

```diff
@@ -82,10 +84,12 @@ export const httpApiKeyAuthMiddleware =
   async (args) => {
     if (!HttpRequest.isInstance(args.request)) return next(args);
 
+    const apiKey = pluginConfig.apiKey && await pluginConfig.apiKey();
+
     // This middleware will not be injected if the operation has the @optionalAuth trait.
     // We don't know if we're the only auth middleware, so let the service deal with the
     // absence of the API key (or let other middleware do its job).
```

Comment:
couple of updates needed to the documentation of this method.
1. `Must include API key` seem out dated
2. the param are incorrect from the method definition

Reply:
done